react/src/hooks/features/useFeatures.ts

```diff
@@ -143,9 +146,35 @@ export const useCurrentFeatures = (): CurrentFeaturesResult => {
       ? queryClient.getQueryState(latestQuery[0])
       : null;
 
+  // Use a ref to keep track of the previous data
+  const prevDataRef = useRef<FeatureCollection | undefined>(undefined);
+  const prevStateRef = useRef<{ pending: boolean; error: boolean }>({
+    pending: false,
+    error: false,
+  });
```

Comment:
Just curious, I wonder if react query's `keepPreviousData` could be of use here?

Reply:
Good idea; that got me working on different approaches. I tried that way and some others but it got tricky with this due to all the different cases and uses, so I ended up keeping some state (with zustand) that could provide data/status for useCurrentFeatures. While maybe not the best (or final) solution, now all the use scenarios are fixed. I couldn't manage getting all the cases to work with react-query and my half-baked interpretation of checking all inactive/active queries.

Reply:
Sounds good. I've had good luck with `ensureQueryData`, prefetching, and suspense queries to ensure data is present regardless of status, but not sure if that's applicable here off the bat